### PR TITLE
Add main interfaces for throttler storage

### PR DESCRIPTION
**Pull request by mayboroda**

### CODE REVIEW

File: src/main/scala/com/wix/ocicat/storage/ThrottlerStorage.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+   * In case invocation of the key `A` is done during next window (for example `W2`) we have to have this state:
+   * ```
+   *  key:A; window:W2; expired:CurrentTime + window; amount: 1
+   * ```
+   *
+   * @param key throttler key
+   * @param rate rate configuration of a throttler
+   * @return capacity of a throttler key of a current tick
+   */
+  def incrementAndGet(key: A, rate: Rate): F[ThrottlerCapacity[A]]
```
Comment (yarhrn) [inline src/main/scala/com/wix/ocicat/storage/ThrottlerStorage.scala:34]: I am not sure generic A here is a right choice taking into account possible remote nature of storage.

File: src/main/scala/com/wix/ocicat/storage/InMemoryStorage.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import java.util.concurrent.TimeUnit
+
+import cats.effect.{Clock, Sync}
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import com.wix.ocicat.Rate
+
+class InMemoryStorage[F[_], A](memory: Ref[F, Map[String, InMemoryValue]], clock: Clock[F])(implicit F: Sync[F]) extends ThrottlerStorage[F, A] {
+
+  private def getPrimaryKey(key: A, tick: Long): String = s"${key}_$tick"
```
Comment (yarhrn) [inline src/main/scala/com/wix/ocicat/storage/InMemoryStorage.scala:12]: This is not safe for any A.
```
case class A(var b: B)
case class B(var a: A)

val a = A(b = null)
val b = B(a = null)
a.b = b
b.a = a
a.toString // <- StackOverflowError here
```

File: src/main/scala/com/wix/ocicat/storage/ThrottlerStorage.scala (hunk 1 above)
Comment (mayboroda) [reply src/main/scala/com/wix/ocicat/storage/ThrottlerStorage.scala:34]: You suggesting to use something like pure `String` or `Serializable`?

File: src/main/scala/com/wix/ocicat/storage/InMemoryStorage.scala (hunk 2 above)
Comment (mayboroda) [reply src/main/scala/com/wix/ocicat/storage/InMemoryStorage.scala:12]: Agree. I think these two comments are linked.

File: src/main/scala/com/wix/ocicat/storage/ThrottlerStorage.scala (hunk 1 above)
Comment (yarhrn) [reply src/main/scala/com/wix/ocicat/storage/ThrottlerStorage.scala:34]: Lets put here string, I believe this is best solution to not deal with serialization

**Pull request merged**